designs/admin-api-health-status.md

```diff
@@ -0,0 +1,36 @@
+---
+state: draft | in-progress | complete
+---
+
+# Add Health and Status Endpoints to Admin API
+
+## Summary
+
+In addition to the available methods in the Admin API laid out [here](https://nodered.org/docs/api/admin/methods/), this document proposes the creation of `health` and `status` endpoints. These endpoints would be useful for application monitoring but could find broader use as well.
+
+While users can add this functionality manully to their own instances of Node-RED, this feature set feels sufficiently primitive and general that it would be useful to expose it as part of the Admin API.
+
+## Authors
+
+- [@szep](https://discourse.nodered.org/u/szep)
+
+## Details
+
+Two endpoints would be added to the Admin API:
+
+1. `GET /health` -- on success returns status code 200 and payload `{"status": "ok"}`
```

Comment:
this is for the overall Node-RED service ? - so yes agree 200 OK fine. (maybe if we want to be detailed could add one for starting and stopping - obviously no response when actually stopped)

Reply:
I think that's a great idea!

Reply:
I think the `/health` end point will need to be configurable - both in terms of whether the endpoint is enabled at all, and what url it is exposed on.

We also have two separate route handlers in Node-RED - the `admin` side and the `node` side. We need to decide which handler these end-points are on. They are probably `admin` routes - although we know some users will disable the admin route handler entirely for their production system.

As there will be existing flows that serve up their own `/health` end point, we may have to disable it by default - and allow it to be enabled via the settings file.